lib/api/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+#![doc(html_favicon_url = "https://wasmer.io/static/icons/favicon.ico")]
+#![doc(html_logo_url = "https://avatars3.githubusercontent.com/u/44205449?s=200&v=4")]
+
+//! TODO: Write high value, high-level API intro docs here
+//! Intro/background information
+//!
+//! quick links to places in this document/other crates/standards etc.
+//!
+//! example code, link to projects using it
+//!
+//! more info, what to do if you run into problems
+
+/// Commonly used types and functions.
+pub mod prelude {
+    pub use crate::module::*;
+    pub use wasmer_runtime_core::instance::{DynFunc, Instance};
+    pub use wasmer_runtime_core::memory::Memory;
+    pub use wasmer_runtime_core::table::Table;
+    pub use wasmer_runtime_core::Func;
+    pub use wasmer_runtime_core::{func, imports};
+}
+
+pub mod module {
+    //! Types and functions for WebAssembly modules.
+    //!
+    //! # Usage
+    //! ## Create a Module
+    //!
+    //! ```
+    //! ```
+    //!
+    //! ## Get the exports from a Module
+    //! ```
+    //! # use wasmer::*;
+    //! # fn get_exports(module: &Module) {
+    //! let exports: Vec<ExportDescriptor> = module.exports().collect();
+    //! # }
+    //! ```
+    // TODO: verify that this is the type we want to export, with extra methods on it
+    pub use wasmer_runtime_core::module::Module;
+    // should this be in here?
+    pub use wasmer_runtime_core::module::{ExportDescriptor, ExportKind, Import, ImportType};
+    // TODO: implement abstract module API
+}
+
+pub mod memory {
+    //! Types and functions for Wasm linear memory.
+    pub use wasmer_runtime_core::memory::{Atomically, Memory, MemoryView};
+}
+
+pub mod wasm {
+    //! Various types exposed by the Wasmer Runtime.
+    //!
+    //! TODO: Add index with links to sub sections
+    //!
+    //! # Globals
+    //!
+    //! # Tables
+    pub use wasmer_runtime_core::global::Global;
+    pub use wasmer_runtime_core::module::{ExportDescriptor, ExportKind, Import, ImportType};
+    pub use wasmer_runtime_core::table::Table;
+    pub use wasmer_runtime_core::types::{
+        FuncSig, GlobalDescriptor, MemoryDescriptor, TableDescriptor, Type, Value,
+    };
+}
+
+pub mod import {
+    //! Types and functions for Wasm imports.
+    pub use wasmer_runtime_core::module::{Import, ImportType};
+    pub use wasmer_runtime_core::{func, imports};
+}
+
+pub mod export {
+    //! Types and functions for Wasm exports.
+    pub use wasmer_runtime_core::module::{ExportDescriptor, ExportKind};
+}
+
+pub mod units {
+    //! Various unit types.
+    pub use wasmer_runtime_core::units::{Bytes, Pages};
+}
+
+pub mod types {
+    //! Types used in the Wasm runtime and conversion functions.
+    pub use wasmer_runtime_core::types::*;
+}
+
+pub mod error {
+    //! Various error types returned by Wasmer APIs.
+    pub use wasmer_runtime_core::error::*;
+}
+
+pub use prelude::*;
+
+/// Idea for generic trait; consider rename; it will need to be moved somewhere else
+pub trait CompiledModule {
```

Comment:
I think this trait should be named `Module`. We can put this contents on the `wasm-common` crate I was working on.

Reply:
Sure, I agree that it's not really in-scope for this.  Though we may run into issues with usability if it has the same name as a concrete type that implements it.